script/testing/micro_bench/artifact_processor.py

```diff
@@ -0,0 +1,165 @@
+        min_build = ref_data_source.get("min_build", 0)
+        status_filter = ref_data_source.get("status_filter")
+        for artifact in jenkins.get_artifacts(folders, project, branch, min_build, status_filter):
+            gbench_run_results = GBenchRunResult(artifact)
+            self.add_artifact(gbench_run_results)
+            # Determine if we have enough history. Stop collecting information if we do
+            if self.has_min_history():
+                return
+
+    def has_min_history(self):
+        """ check whether all the collected artifacts have at least the minimum
+            number of results.
+        """
+        if not self.required_num_results:
+            LOG.debug("required_num_results is not set???")
+            return False
+        if len(self.artifacts) == 0:
+            LOG.debug("No artifacts available")
+            return False
+
+        for key in self.artifacts.keys():
+            artifact = self.artifacts.get(key)
+            (suite_name, test_name) = key
+            LOG.debug("# of artifacts for {SUITE}.{TEST}: {NUM_ARTIFACTS} [required={REQUIRED_ARTIFACTS}]".format(
+                SUITE=suite_name, TEST=test_name, NUM_ARTIFACTS=artifact.get_num_results(), REQUIRED_ARTIFACTS=self.required_num_results))
+            if artifact.get_num_results() < self.required_num_results:
+                return False
+        return True
+
+    def add_artifact(self, gbench_run_result):
+        """ Add an artifact file to the list of files to be used
+        for computing summary statistics
+        """
+        # iterate over the GBBenchResult objects
+        for key, bench_result in gbench_run_result.benchmarks.items():
+            # add to a GBBenchResultProcessor
+            historical_results = self.artifacts.get(
+                key, GBenchHistoricalResults(*key))
+            self.artifacts[key] = historical_results
+            if not (self.required_num_results and historical_results.get_num_results() >= self.required_num_results):
+                historical_results.add_gbench_test_result(bench_result)
+        return
+
+    def get_comparison_for_publish_result(self, bench_name, gbench_result, lax_tolerance=None):
+        """ create and return a dict comparing the historical benchmark results
+        to the benchmark result passed in. This will format the results in a
+        way that the performance storage service will understand. The main 
+        difference between this method and get_comparison is stdev_throughput.
+        """
+        initial_comparison = self.get_comparison(bench_name, gbench_result, lax_tolerance)
+        fields = ['suite', 'test', 'throughput', 'tolerance',
+                  'status', 'iterations', 'ref_throughput', 'num_results']
+
+        publishable_comparison = {
+            key: value for key, value in initial_comparison.items() if key in fields}
+
+        test_suite, test_name = initial_comparison.get('suite'), initial_comparison.get('test')
```

Comment:
Looks like `test_suite` and `test_name` are not used in the function at all.

Reply:
oops